### ISSUE TITLE
Not pulling from Trakt when adding new media to Plex

### ISSUE THREAD

**Issue by DaveAlton** [trakt/Plex-Trakt-Scrobbler#351]
To reproduce the problem, I have Trakt installed, and working. If I load a new series into Plex that I have currently marked as watched in Trakt, it doesn't mark any of the episodes as watched. When running a full sync or quick pull it doesn't work, but if I do a regular pull, it works great. The problem for me is that I want this to automatically sync whether I have watched an item as soon as I add it to Plex. The Trakt channel currently has a box for just that: "Sync - Run after library updates".

This worked for me in v0.9.1.29, but I have just updated. It may be caused by some of my settings though. I'm currently running v1.0.2.0

The metadata agent I'm currently using on my tv shows library is TheTVDb.

MacOS 10.11.5
Plex Version 2.7.4
Client is just my laptop.

Here are my current settings. I've removed the pin for safety reasons.

```
<?xml version='1.0' encoding='utf-8'?>
<PluginPreferences>
  <sync_idle_defer>false</sync_idle_defer>
  <sync_watchlist/>
  <pin>PIN</pin>
  <scrobble_duplication_period/>
  <filter_networks>*</filter_networks>
  <scrobble_clients/>
  <level_pyemitter>1</level_pyemitter>
  <sync_idle_delay/>
  <start_scrobble>true</start_scrobble>
  <sync_ratings/>
  <sync_run_library>true</sync_run_library>
  <filter_sections>*</filter_sections>
  <sync_personal_lists/>
  <scrobble_names/>
  <activity_mode/>
  <sync_clean_collection>false</sync_clean_collection>
  <matcher/>
  <level_libraries>1</level_libraries>
  <sync_liked_lists/>
  <sync_ratings_conflict/>
  <language/>
  <sync_playback>1</sync_playback>
  <sync_run_interval>1</sync_run_interval>
  <sync_collection/>
  <sync_watched>1</sync_watched>
  <level_peewee>1</level_peewee>
  <sync_instant_actions>true</sync_instant_actions>
  <level_plugin>1</level_plugin>
</PluginPreferences>
```

Thanks so much for your help

**Comment by fuzeman** [comment]
Currently automatic syncs will only import changes to your Trakt.tv profile since the last sync, this will cause what you have described with watched states, ratings, etc.. not being imported for new items that are added to Plex.

--------

The best solution would probably be detecting new items that have been added to Plex, and just ensuring an initial "Pull" sync is run on them.

I'll try find some time later this month (or early next month) to work on this, and will update this issue when there is a new release available to test.

**Comment by DaveAlton** [comment]
That would be great, thanks so much for your help!

**Comment by geogolem** [comment]
I have a request that is somewhat the opposite of this request -- or different. Perhaps related (since it has to do with adding new items to the plex library) I wasn't sure if I should create a new issue/feature request or just post here. I decided I would post it here, and then it will eventually get directed to the most appropriate place.

Before I get into what I want in this new "feature" I would first like to ask a few questions so that I could better clarify what some existing feature is doing or intended for.

Specifically I am referring to "sync - collection" {Disabled, Push, Pull, Synchonize Push+Pull}

I have this setting set to, PUSH. 

My understanding is that this will add anything that exists in my plex library but doesnt already exist in my trakt collection to my trakt collection. Is my understanding correct? I believe this will never remove anything from my trakt collection. 

The next tick box "sync - clean collection" will remove items from the trakt library that have been deleted from my plex library. Is this correct?

What does sync - collection: PULL do? Plex builds its own library based on what movies it finds stored on the HDD.. Wouldn't a PULL from the trakt for the collection list be kind of pointless? Wont PLEX just ignore anything it doesnt find on the storage corresponding to its libraries? Similarly then, what exactly does the "synchronize Pull+Push" setting on sync - collection do?

In any case, I am simply using a sync - collection (PUSH) and that mostly does what I want it to do when combined with the sync - clean collection.. If my understanding is correct. Im just curious what the other possibilities actually do?

-----------------
Onto my request or wish....

When I add a new movie to plex... the sync - collection (PUSH) kicks in and adds this movie to my trakt collection (if it doesnt already exist) 

I would also like it to be added to my watchlist if it has not been marked as watched (or if it is not in my  trakt watched list)...

i.e. any movie added to my collection should be added to my watch list unless ive already watched that movie..  

so it would be nice if there was a tickbox right under sync - collection similar to sync - clean collection that said "add unwatched items to watchlist)... or something along those lines..

Does what I am requesting make sense?

so if i add a new item to my plex library... I want that to get added to my trakt collection... and if I havent already seen that movie (according to trakt watched list) add to it my watchlist because that list keeps track of what I want to watch and I plan to watch everything in my collection that I havent already seen..

Please let me know if you need further clarification. I dont know how difficult it would be to add such a minor option... but it would be very beneficial for me - not sure about others.

**Comment by fuzeman** [comment]
Sorry for not looking at this issue yet, have been very busy the last few months working on some other projects. Will try find some time this week to work on it :smiley:

----------------------
@geogolem 

> Specifically I am referring to "sync - collection" {Disabled, Push, Pull, Synchonize Push+Pull}
> 
> I have this setting set to, PUSH.
> 
> My understanding is that this will add anything that exists in my plex library but doesnt already exist in my trakt collection to my trakt collection. Is my understanding correct? I believe this will never remove anything from my trakt collection.

Yes

> The next tick box "sync - clean collection" will remove items from the trakt collection that have been deleted from my plex library. Is this correct?

Yes

> What does sync - collection: PULL do? Plex builds its own library based on what movies it finds stored on the HDD.. Wouldn't a PULL from the trakt for the collection list be kind of pointless? Wont PLEX just ignore anything it doesnt find on the storage corresponding to its libraries? Is it possible that movies deleted from trakt will be deleted from the Plex library... if this is on? Similarly then, what exactly does the "synchronize Pull+Push" setting on sync - collection do?

It doesn't do anything, see: https://github.com/trakt/Plex-Trakt-Scrobbler/blob/master/Trakttv.bundle/Contents/Libraries/Shared/plugin/sync/handlers/collection/pull.py

I would like to change the values listed under the "Sync - Collection" option, but there is a chance changing them will cause unexpected behavior for existing users *(plugin framework issue)*. Will need to do some more testing to see if we can change the drop-down option values without breaking anything...

> When I add a new movie to plex... the sync - collection (PUSH) kicks in and adds this movie to my trakt collection (if it doesnt already exist)
> 
> I would also like it to be added to my watchlist if it has not been marked as watched (or if it is not in my trakt watched list)...
> 
> i.e. any movie added to my collection should be added to my watch list unless ive already watched that movie..
> 
> so it would be nice if there was a tickbox right under sync - collection similar to sync - clean collection that said "add unwatched items to watchlist)... or something along those lines..
> 
> Does what I am requesting make sense?

👍 Sounds like a useful feature. I've created https://github.com/trakt/Plex-Trakt-Scrobbler/issues/370 to track this feature request.

**Comment by DaveAlton** [comment]
Thanks! Just the fact that you’re going to look at it at some point is awesome. I really appreciate it!

On Mon, Nov 14, 2016 at 4:57 PM Dean Gardiner

<
mailto:Dean Gardiner <notifications@github.com>
> wrote:

a, pre, code, a:link, body { word-wrap: break-word !important; }

Sorry for not looking at this issue yet, have been very busy the last few months working on some other projects. Will try find some time this week to work on it

😃

https://github.com/geogolem

Specifically I am referring to "sync - collection" {Disabled, Push, Pull, Synchonize Push+Pull}

I have this setting set to, PUSH.

My understanding is that this will add anything that exists in my plex library but doesnt already exist in my trakt collection to my trakt collection. Is my understanding correct? I believe this will never remove anything from my trakt collection.

Yes

The next tick box "sync - clean collection" will remove items from the trakt collection that have been deleted from my plex library. Is this correct?

Yes

What does sync - collection: PULL do? Plex builds its own library based on what movies it finds stored on the HDD.. Wouldn't a PULL from the trakt for the collection list be kind of pointless? Wont PLEX just ignore anything it doesnt find on the storage corresponding to its libraries? Is it possible that movies deleted from trakt will be deleted from the Plex library... if this is on? Similarly then, what exactly does the "synchronize Pull+Push" setting on sync - collection do?

It doesn't do anything, see:
https://github.com/trakt/Plex-Trakt-Scrobbler/blob/master/Trakttv.bundle/Contents/Libraries/Shared/plugin/sync/handlers/collection/pull.py

I would like to change the values listed under the "Sync - Collection" option, but there is a chance changing them will cause unexpected behavior for existing users

(plugin framework issue)

. Will need to do some more testing to see if we can change the drop-down option values without breaking anything...

When I add a new movie to plex... the sync - collection (PUSH) kicks in and adds this movie to my trakt collection (if it doesnt already exist)

I would also like it to be added to my watchlist if it has not been marked as watched (or if it is not in my trakt watched list)...

i.e. any movie added to my collection should be added to my watch list unless ive already watched that movie..

so it would be nice if there was a tickbox right under sync - collection similar to sync - clean collection that said "add unwatched items to watchlist)... or something along those lines..

Does what I am requesting make sense?

👍

Sounds like a useful feature. I've created
https://github.com/trakt/Plex-Trakt-Scrobbler/issues/370
to track this feature request.

—

You are receiving this because you authored the thread.

Reply to this email directly,
https://github.com/trakt/Plex-Trakt-Scrobbler/issues/351#issuecomment-260513869
, or

**Comment by fuzeman** [comment]
I've got this partially implemented on the [feature/351](https://github.com/fuzeman/Plex-Trakt-Scrobbler/tree/feature/351) branch, will be making some more improvements and testing it a bit more this weekend.

**Comment by fuzeman** [comment]
Merged into the [develop](https://github.com/trakt/Plex-Trakt-Scrobbler/archive/develop.zip) branch, will be included in the next release.

**Comment by DaveAlton** [comment]
Thanks so much! This will be so helpful!

**Comment by DaveAlton** [comment]
Downloaded and installed the develop version and it works great. So pleased. This plugin went from something that was useful and mostly automated to one that is useful and fully automated. Thanks so much!

**Comment by fuzeman** [comment]
Released in [v1.2.0](https://github.com/trakt/Plex-Trakt-Scrobbler/releases/tag/v1.2.0)
